Chatkit/Model/User.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+import CoreData
+
+public struct User {
+    
+    // MARK: - Properties
+    
+    public let identifier: String
+    public let name: String?
+    public let avatar: URL?
+    public let presenceState: PresenceState
+    public let metadata: Metadata?
+    public let createdAt: Date
+    public let updatedAt: Date
+    
+    let objectID: NSManagedObjectID
+    
+    // MARK: - Initializers
+    
+    init(identifier: String, name: String?, avatar: URL?, presenceState: PresenceState, metadata: Metadata?, createdAt: Date, updatedAt: Date, objectID: NSManagedObjectID) {
```

Comment:
I think that with `struct`'s we can even leave the `init` out as the compiler will create one for us.

Reply:
That is true. However, there is one nasty side effect of using auto-generated initializers in Swift. Changing the order of properties in the file changes the order of parameters in the initializer.

Example:

#### Variant A

```swift
public struct User {
    public let identifier: String
    public let name: String
}
```

Generates:

```swift
public init(identifier: String, name: String) {
    self.identifier = identifier
    self.name = name
}
```

#### Variant B

```swift
public struct User {
    public let name: String
    public let identifier: String
}
```

Generates:

```swift
public init(name: String, identifier: String) {
    self.name = name
    self.identifier = identifier
}
```

This is a subtle change, but it has the potential to break the public interface. That is why I would prefer to be extra cautious and generate initializers manually.

There is also one more problem here. Auto-generated initializer would be defined as `public` and we do not want anyone to instantiate our model outside our SDK.